Mathlib/Algebra/MonoidAlgebra/NoZeroDivisors.lean

```diff
@@ -46,6 +50,15 @@ namespace AddMonoidAlgebra
 
 open Finsupp
 
+instance {L σ : Type*} [LinearOrder σ] [LinearOrder L] [AddGroup L]
+    [ContravariantClass L L (· + ·) (· ≤ ·)]
+    [CovariantClass L L (Function.swap (· + ·)) (· ≤ ·)] :
+    UniqueSums (σ →₀ L) := show UniqueSums ((Lex (σ →₀ L))) from
```

Comment:
I think we can easily show `UniqueProds (G × H)` from `UniqueProds G` and `UniqueProds H`, and by induction on `Pi` (finitely many factors), `Finsupp` and `DFinsupp`. However, unlike linear orderability, I think `UniqueProds` probably doesn't transfer to direct product (`Pi`) with infinitely many factors, but I'm yet to find a counterexample.

Reply:
I do not know the answer to the question about infinite products.  However, these notions are very close and subtly different from torsion-freeness.

Kaplanski's conjecture asserts that torsion-freeness for a group `G` should be equivalent to no non-zero zero-divisors in `k[G]`.

Maybe there is some room for an easy `Pi`-counterexample using Monoids/Semigroups?  However, a counterexample with a group may be hard...

Reply:
Kaplansky's conjecture is falsified but it's still unknown whether unique product <=> no zero divisors IIUC, and that's a hint that it should work for infinite direct products, which is what I have proven.